src/components/data-exploration/heatmap/HeatmapPlot.jsx

```diff
@@ -149,29 +162,14 @@ const HeatmapPlot = (props) => {
   }, [louvainClusterCount]);
 
   useEffect(() => {
-    setMaxCells(Math.floor(width * 0.8));
-  }, [width]);
-
-  const handleMouseOver = (...args) => {
-    if (args.length < 2) {
-      return;
-    }
+    dispatch(updateCellInfo({ cellName: cellHighlight }));
+  }, [cellHighlight]);
 
-    if ('x' in args[1] && 'y' in args[1]) {
-      hoverCoordinates.current = {
-        x: args[1].x,
-        y: args[1].y,
-      };
-    }
-
-    if (!args[1].datum) {
-      return;
-    }
+  useEffect(() => {
+    if (!heatmapData) return;
 
-    const { cellId: cellName } = args[1].datum;
-
-    dispatch(updateCellInfo({ cellName }));
-  };
+    setVitessceData(heatmapData);
+  }, [heatmapData]);
 
```

Comment:
for more unit test inspirations -- code coverage on my VS code is telling me that these are not tested, so you can write unit tests for them